src/frontend/js/widgets/Dashboard/components/DashboardItem/Enrollment/ProductCertificateFooter/index.spec.tsx

```diff
@@ -147,10 +147,40 @@ describe('<ProductCertificateFooter/>', () => {
       CertificateFactory({ id: order.certificate_id }).one(),
     );
     render(<ProductCertificateFooter product={product} enrollment={enrollment} />);
-    expect(await screen.findByRole('button', { name: 'Download' })).toBeInTheDocument();
+    expect(await screen.getByRole('button', { name: 'Download' })).toBeInTheDocument();
     expect(screen.queryByTestId('PurchaseButton__cta')).not.toBeInTheDocument();
   });
 
+  it('should not display purchase button for a course run with submitted order.', async () => {
+    const order = OrderEnrollmentFactory({
+      certificate_id: undefined,
+      product_id: product.id,
+      state: OrderState.SUBMITTED,
+    }).one();
+    const enrollment = EnrollmentFactory({
+      orders: [order],
+      course_run: CourseRunFactory({ course }).one(),
+    }).one();
+    render(<ProductCertificateFooter product={product} enrollment={enrollment} />);
+    expect(await screen.queryByRole('button', { name: 'Download' })).not.toBeInTheDocument();
+    expect(screen.queryByTestId('PurchaseButton__cta')).not.toBeInTheDocument();
+  });
+
+  it('should display purchase button for a course run with pending order.', async () => {
+    const order = OrderEnrollmentFactory({
+      certificate_id: undefined,
+      product_id: product.id,
+      state: OrderState.PENDING,
+    }).one();
+    const enrollment = EnrollmentFactory({
+      orders: [order],
+      course_run: CourseRunFactory({ course }).one(),
+    }).one();
+    render(<ProductCertificateFooter product={product} enrollment={enrollment} />);
+    expect(await screen.queryByRole('button', { name: 'Download' })).not.toBeInTheDocument();
```

Comment:
Hmmm `queryBy` doesn't return a promise. You should use `findByRole` if you want a promise. But in your case it should not work.